recipes/zulu-openjdk/all/conanfile.py

```diff
@@ -0,0 +1,59 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os, platform, glob
+
+
+class ZuluOpenJDK(ConanFile):
+    name = "zulu-openjdk"
+    url = "https://github.com/conan-io/conan-center-index/"
+    description = "A OpenJDK distribution"
+    homepage = "https://www.azul.com"
+    license = "https://www.azul.com/products/zulu-and-zulu-enterprise/zulu-terms-of-use/"
+    topics = ("java", "jdk", "openjdk")
+    settings = "os", "arch"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _jni_folder(self):
+        folder = {"Linux": "linux", "Darwin": "darwin", "Windows": "win32"}.get(platform.system())
+        return os.path.join("include", folder)
+
+    def configure(self):
+        # Checking against self.settings.* would prevent cross-building profiles from working
+        # TODO, I think for the new, 2 profile builds, this can be done ...
+        if tools.detected_architecture() != "x86_64":
```

Comment:
`tools.detected_architecture()` is not documented. What breaks if you use self.settings.os ?

Reply:
That's the problem when PRs are so long in the queue, details get forgotten

But I think the reason was, that self setting.os had an issue when it was used for cross compiling, the old way, with 1 profile (os_build) and the jdk was used as a build requirement

Reply:
ok, so you should definitely use `self.settings.os` and `self.settings.arch`, as is explained in https://docs.conan.io/en/latest/systems_cross_building/cross_building.html?highlight=os_build#conan-v1-24-and-newer

Reply:
there is still the _experimentell_ note to `--profile::build  and --profile::host` 
when this is gone I will happily change that, here and in my ios-cmake PR 

however, I think also a good change would be to document `tools.detected_architecture()` and make it official, because it might be useful on other places as well

Reply:
I'm not sure why this is still considered experimental, especially when it is written on the same page `We ask you to use the previous approach for Conan 1.24 and newer` (previous approach being the one with 2 profiles), and `avoid any specific modification of your recipes to make them work as build requirements in a cross building scenario.` which is exactly what you are doing here.
Existing recipes which had a chance to be updated are using this approach : [ninja](https://github.com/conan-io/conan-center-index/blob/master/recipes/ninja/1.10.x/conanfile.py), [cmake](https://github.com/conan-io/conan-center-index/blob/master/recipes/cmake/3.x.x/conanfile.py) and probably others
Regarding the documentation of `tools.detected_architecture()` : https://github.com/conan-io/docs/issues/1035

Reply:
ok